core/storage/translation/gae_models.py

```diff
@@ -0,0 +1,141 @@
+    [models.NAMES.base_model])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MachineTranslatedTextModel(base_models.BaseModel):
+    """Model for storing machine translations. Model instances have a key
+    generated from the source and target language codes, followed by a sha-1
+    hash of the origin text formated as follows:
+        '[source_language_code]:[target_language_code]:[hashed_text]'
+    See MachineTranslatedTextModel._generate_id() below for details."""
+
+    # The untranslated text.
+    origin_text = datastore_services.TextProperty(required=True, indexed=False)
+    # The language code of the untranslated text.
+    source_language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The language code of the translation.
+    target_language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The translation.
+    translated_text = datastore_services.TextProperty(
+        required=True, indexed=False)
+
+    @classmethod
+    def create(
+            cls, source_language_code, target_language_code, origin_text,
+            translated_text):
+        """Creates a new MachineTranslatedTextModel instance and returns its
+        ID.
+
+        Args:
+            source_language_code: str. The language code of the untranslated
+                text.
+            target_language_code: str. The language code of the translation.
+            origin_text: str. The untranslated text.
+            translated_text: str. The translation.
+
+        Returns:
+            MachineTranslatedTextModel. The newly created
+            MachineTranslatedTextModel instance.
+        """
+
+        entity_id = cls._generate_id(
+            source_language_code, target_language_code, origin_text)
+        translation_entity = cls(
+            id=entity_id,
+            source_language_code=source_language_code,
+            target_language_code=target_language_code,
+            origin_text=origin_text,
+            translated_text=translated_text)
+        translation_entity.put()
+        return entity_id
+
+    @classmethod
+    def _generate_id(
+            cls, source_language_code, target_language_code, origin_text):
+        """Generates a key for the instance of MachineTranslatedTextModel
+        class in the required format with the arguments provided.
+
+        Args:
+            source_language_code: str. The language code of the untranslated
+                text.
+            target_language_code: str. The language code of the translation.
+            origin_text: str. The untranslated text.
+
+        Returns:
+            str. The generated ID for this entity of the form
+            '[source_language_code]:[target_language_code]:[hashed_text]'.
+        """
+        # max_length is 450, less than maximum key length (500) to allow room
+        # for language_codes.
+        hashed_text = utils.convert_to_hash(origin_text, 50)
```

Comment:
Should the max length be set to 450 like the comment says? Also nit: prefer to capitalize the comment sentence, especially if you're adding a period to make it a "real" sentence".

Reply:
I forgot to remove this comment after addressing [Sean's comment](https://github.com/oppia/oppia/pull/12515#discussion_r613610177).